src/storage.rs

```diff
@@ -14,20 +24,44 @@ impl Storage {
     /// let mut stor = Builder::new().build();
     /// let res = stor.init();
     /// ```
-    pub fn init(&mut self) -> Result<(), ()> {
+    ///
+    /// Storage works in dir provided to builder. If dir not exist,
+    /// creates it, otherwise tries init dir as existing storage.
+    pub fn init(&mut self) -> io::Result<()> {
+        // @TODO implement work dir validation
+        self.prepare_work_dir()?;
+        let wd = Path::new(&self.config.work_dir);
+        let files_in_work_dir: Vec<_> = fs::read_dir(wd)?
```

Comment:
doing twice read_dir

Reply:
we can return file list from prepare_work_dir

Reply:
Fixed. Check existence with [Path::exists(_)](https://doc.rust-lang.org/std/path/struct.Path.html#method.read_dir)